lib/utils/job-utils/command-util.js

```diff
@@ -96,11 +96,22 @@ function commandUtilFactory(
                     result.source + ', ' + result.error,
                     { error: result });
             } else if (result.store) {
-                return waterline.catalogs.create({
+                var dataToSave = {
                     node: self.nodeId,
                     source: result.source || 'unknown',
                     data: result.data
-                });
+                };
+                if (self.updateExisting) {
+                    return waterline.catalogs.update(
```

Comment:
@yyscamper Good idea. I'll change the code to create a new catalog if it doesn't exist...